### DISCUSSION TITLE
What are the changes behind v2 and v3?

### DISCUSSION THREAD

**Discussion by tnielens** [pityka/saddle#385] [General]
Hi @pityka, 
I'm curious about the changes that lead your fork to v2 and v3. What incompatible changes were introduced in those versions?

**Comment by pityka** [comment]
Only the sbt-version-policy plugin in the build. I thought it is better to signal the strict enforcement of the version scheme policy with a new major.

**Reply by tnielens** [reply]
Alright makes sense 👍. It might be worth documenting this somewhere. I think users care about the release version scheme of projects, how much care is put in guaranteeing backward binary compatibility and how long major versions are maintained. As an example of project I look up to regarding their version management: https://http4s.org/versions.html

**Reply by pityka** [reply]
Having RCs and milestones and parallel stable tracks is very good for the library clients but i have no resources (time) to manage all that unfortunately. 

Also saddle adoption (at least this fork) is pretty low. Idk if anybody is using it except me. 

So I am with a simple head only versioning with mima guaranteeing that the "abi" and the version string is consistent, but otherwise I can't thrive for very long support. If I need a feature which can't be implemented in a backward compatible manner then I would need to bump the major. That said I did not contribute new feature to saddle in 1 year so I expected that it is hm finished and for that reason I pulled in sbt version policy and started the v3 branch.

**Reply by tnielens** [reply]
I sure understand the work necessary to maintain several major versions in parallel and that it is in end up to the maintainer. My suggestion is to describe briefly the differences between v1, v2, v3 in the doc somewhere or the README file as that's something I'm usually interested in when looking at a new library.
